packages/access-client/src/agent-use-cases.js

```diff
@@ -263,6 +263,12 @@ export async function authorizeWithPollClaim(access, email, opts) {
     ...opts,
     expectAuthorization,
   })
+  // Claim delegations here because we will need an ucan/attest from the service to
+  // pair with the session delegation we just claimed to make it work.
+  // This is a bit duplicative because we're already invoking `access/claim` in the
+  // wait function above, but we don't do space inference there so do it one more time
+  // for consistency with authorizeWithSocket.
+  await claimAccess(access, access.issuer.did(), { addProofs: true })
```

Comment:
Can you either define this as a new thing, or else wherever you are tempted to use this, due `authorizeWithPollClaim().then(() => claimAccess(...))`

Reply:
I have a strong preference for `authorizeWithPollClaim` and `authorizeWithSocket` both behaving the same way here - either they should both automatically `claimAccess` or both not do that. Since `authorizeWithSocket` is already in use in both clients and `authorizeWithPollClaim` is unused I'd like to stick with `authorizeWithSocket`'s semantics for now.

I know you would like to simplify this layer and push responsibility for calling `claimAccess` up to the clients and I'm generally supportive of that, but I think we should open an issue for that refactoring and address it once we're out of crunch mode. 

wdyt?

Reply:
fwiw I'm also open to defining a new thing, but I think it should be named differently - ie, I think `authorize{WithSocket|WithPollClaim}` should behave the same way - if you'd like to preserve a use case that doesn't `claimAccess` that's fine but lets call it something different to distinguish it from the existing functionality.

Reply:
one more thought on this - I could call `claimAccess` in `react-keyring/src/providers/Keyring.tsx` but fwiw if feels awkward to be dealing with lower level details like the issuer did - ie, I could go from this:

```tsx
      await authorizeWithPollClaim(agent, email, { signal: controller.signal })
```

to this:

```tsx
      await authorizeWithPollClaim(agent, email, { signal: controller.signal })
      await accessClaim(agent, agent.issuer.did(), {addProofs: true})
```

but that gets this React module in the business of thinking about issuer DIDs and whether it should "add proofs" - I'd rather encapsulate those details down here in the use cases, which suggests to me that we should have some function in here that do what I'm doing for both the websocket and polling cases no matter what.

Reply:
what's important to me is that we maintain there being a way of calling `authorizeWithPollClaim` that does not also do more stuff than that. That way you can compose it with other things. What I don't like about adding more functionality to the function that was there is that it makes the function that was there no longer useful for the case where you don't want to addProofs.

Reply:
>  I could call claimAccess in react-keyring/src/providers/Keyring.tsx but fwiw if feels awkward to be dealing with lower level details like the issuer DID and the addProofs option at that level.

so do it in w3up-client https://github.com/web3-storage/w3ui/issues/295